astra-sim/workload/Workload.cc

```diff
@@ -151,127 +158,146 @@ void Workload::issue_remote_mem(shared_ptr<Chakra::ETFeederNode> node) {
   WorkloadLayerHandlerData* wlhd = new WorkloadLayerHandlerData;
   wlhd->sys_id = sys->id;
   wlhd->workload = this;
-  wlhd->node_id = node->getChakraNode()->id();
-  sys->remote_mem->issue(node->getChakraNode()->tensor_size(), wlhd);
+  wlhd->node_id = node->id();
+  sys->remote_mem->issue(node->tensor_size(), wlhd);
 }
 
 void Workload::issue_comp(shared_ptr<Chakra::ETFeederNode> node) {
-  assert(node->getChakraNode()->node_type() == ChakraNodeType::COMP_NODE);
   hw_resource->occupy(node);
 
   if (sys->roofline_enabled) {
     WorkloadLayerHandlerData* wlhd = new WorkloadLayerHandlerData;
-    wlhd->node_id = node->getChakraNode()->id();
+    wlhd->node_id = node->id();
 
     double operational_intensity =
-        static_cast<double>(node->getChakraNode()->num_ops()) /
-        static_cast<double>(node->getChakraNode()->tensor_size());
+        static_cast<double>(node->num_ops()) /
+        static_cast<double>(node->tensor_size());
     double perf = sys->roofline->get_perf(operational_intensity);
     double elapsed_time =
-        static_cast<double>(node->getChakraNode()->num_ops()) / perf;
-    uint64_t simulated_run_time =
+        static_cast<double>(node->num_ops()) / perf;
+    uint64_t runtime =
         static_cast<uint64_t>(elapsed_time * static_cast<double>(FREQ));
-    sys->register_event(this, EventType::General, wlhd, simulated_run_time);
+    sys->register_event(this, EventType::General, wlhd, runtime);
   } else {
     WorkloadLayerHandlerData* wlhd = new WorkloadLayerHandlerData;
-    wlhd->node_id = node->getChakraNode()->id();
+    wlhd->node_id = node->id();
 
     sys->register_event(
         this,
         EventType::General,
         wlhd,
-        node->getChakraNode()->simulated_run_time());
+        // chakra runtimes are in microseconds and we should convert it into nanoseconds
+        node->runtime() * 1000);
```

Comment:
we should add an assert to make sure the node has a valid runtime.

Reply:
Updated.